plugin/src/com/microsoft/alm/plugin/idea/tfvc/core/TfvcIntegrationEnabler.java

```diff
@@ -0,0 +1,304 @@
+        ShowSettingsUtil.getInstance().showSettingsDialog(project, TFSVcs.TFVC_NAME);
+    }
+
+    private static void showNoWorkspaceDetectedDialog(@Nullable Project project) {
+        ApplicationManager.getApplication().assertIsDispatchThread();
+
+        Messages.showErrorDialog(
+                project,
+                TfPluginBundle.message(TfPluginBundle.KEY_TFVC_WORKSPACE_NOT_DETECTED),
+                TfPluginBundle.message(TfPluginBundle.KEY_TFVC_IMPORT_WORKSPACE_TITLE));
+    }
+
+    private static CompletionStage<AuthenticationInfo> getAuthenticationInfoAsync(@NotNull String serverUrl) {
+        CompletableFuture<AuthenticationInfo> result = new CompletableFuture<>();
+
+        try {
+            // First, try to get the authentication info from existing context:
+            ourLogger.info("Server context lookup for URL: " + serverUrl);
+            ServerContextManager serverContextManager = ServerContextManager.getInstance();
+            ServerContext serverContext = serverContextManager.get(serverUrl);
+            if (serverContext != null) {
+                ourLogger.info("Server context found for URL: " + serverUrl);
+                AuthenticationInfo authenticationInfo = serverContext.getAuthenticationInfo();
+                if (authenticationInfo != null) {
+                    ourLogger.info("Authentication info found for URL: " + serverUrl);
+                    result.complete(authenticationInfo);
+                    return result;
+                }
+            }
+
+            ourLogger.info("Authentication provider pass for URL: " + serverUrl);
+            AuthenticationProvider authenticationProvider = serverContextManager.getAuthenticationProvider(serverUrl);
+            authenticationProvider.authenticateAsync(serverUrl, new AuthenticationListener() {
+                @Override
+                public void authenticating() {
+                    ourLogger.info("Authenticating URL: " + serverUrl);
+                }
+
+                @Override
+                public void authenticated(AuthenticationInfo authenticationInfo, Throwable throwable) {
+                    ourLogger.info("Authentication result for URL " + serverUrl + ": " + (authenticationInfo != null));
+                    if (throwable != null) {
+                        result.completeExceptionally(throwable);
+                        return;
+                    }
+
+                    result.complete(authenticationInfo);
+                }
+            });
+        } catch (Throwable t) {
+            result.completeExceptionally(t);
+        }
+
+        return result;
+    }
+
+    public static CompletionStage<Boolean> importWorkspaceAsync(@Nullable Project project, @NotNull ProgressIndicator indicator, @NotNull Path workspacePath) {
+        Application application = ApplicationManager.getApplication();
+
+        final double totalSteps = 5.0;
```

Comment:
Are all steps has a similar time?

Reply:
According to the best guess we can do here, yes.